### PR TITLE
ext_proc: Use the new mutation rules and new proto

### PR DESCRIPTION
**Pull request by gbrail**
Commit Message: ext_proc: Add support for new configuration to fine-tune which message headers an external processing server is allowed to modify
Additional Description: Adds a new configuration element to ext_proc configuration that gives fine-grained control over how a processor may modify headers, and over what to do if those rules are not followed.
Risk Level: Medium -- all existing functionality should continue to work the same.
Testing: New unit and integration tests.
Docs Changes:
Release Notes:
Platform Specific Features:
Signed-off-by: Gregory Brail <gregbrail@google.com>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
cc @chaoqin-li1123

**Comment by gbrail** [comment]
Hi! Is there anything that you all need from me in order to move this along? I don't know if the bot thinks that we're waiting for me or if we're waiting on someone else...

File: source/extensions/filters/http/ext_proc/ext_proc.cc [hunk 1]
```diff
@@ -512,6 +541,16 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
...
+  } else {
+    // Any other error results in an immediate response with an error message.
+    ENVOY_LOG(debug, "Sending immediate response: {}", processing_status.message());
+    processing_complete_ = true;
+    closeStream();
+    cleanUpTimers();
+    ImmediateResponse invalid_mutation_response;
+    invalid_mutation_response.mutable_status()->set_code(StatusCode::InternalServerError);
+    invalid_mutation_response.set_details(std::string(processing_status.message()));
+    sendImmediateResponse(invalid_mutation_response);
```
Comment (snowp) [inline source/extensions/filters/http/ext_proc/ext_proc.cc:553]: Should we be incrementing a stat for this pathway? Or is that already covered?

**Review by snowp — COMMENTED** [review]
Thanks this looks good to me, one question

Sorry for the review delay! Most of us rely on being assigned to the PR (the review quested is a bit too noisy). You an assign me or whoever by invoking

/assign @snowp 

though normally it's the oncalls job to do that as part of triaging.

File: source/extensions/filters/http/ext_proc/ext_proc.cc (hunk 1 above)
Comment (gbrail) [reply source/extensions/filters/http/ext_proc/ext_proc.cc]: Yes, I think we should increment the "stream_msgs_received" counter here, so I fixed that. Thanks!

**Review by snowp — APPROVED** [review]
Thanks!

**Comment by gbrail** [comment]
This was the last PR needed in order to close https://github.com/envoyproxy/envoy/issues/14789

**Pull request merged**